### PR TITLE
Multi Instance widget core

### PR DESCRIPTION
**Pull request by arjunu**
- [x] Support for multi instantiation
  - Rewrite widget-core to support being able to create and use multiple widget core instances for chatting with multiple bots at the same time.
  - Saved state is per bot, so conversation resume for all the bots under the same domain.
- [x] Return debug data to consumer

### CODE REVIEW

File: src/tests/index.test.js [hunk 1]
```diff
@@ -1,24 +1,59 @@
...
     // rawhide
     expect(
       notificationsChannelURLFromHookURL(
         'https://api.rawhide.canvas.twyla.ai/widget-hook/massive-dynamic/z10'
       )
-    ).toEqual('wss://api.rawhide.canvas.twyla.ai/widget-notifications/massive-dynamic/z10');
+    ).toEqual({
+      botName: 'Z10',
+      notificationsChannelURL:
+        'wss://notification.rawhide.canvas.twyla.ai/widget-notifications/massive-dynamic/z10',
```
Comment (abn) [inline src/tests/index.test.js]: This is a regression. Notification endpoint is deprecated.

File: src/utils.js [hunk 2]
```diff
@@ -1,5 +1,68 @@
+import { CONVERSATION_STARTER, TemplateTypes } from './constants';
```
Comment (abn) [inline src/utils.js]: This seems to be regressing. See https://github.com/twyla-ai/widget-core/commit/4b38728369916318a7b163d0e226126cf30d9806

**Comment by abn** [comment]
@arjunu looks like you did some merges accepting your changes ;)

**Comment by arjunu** [comment]
> @arjunu looks like you did some merges accepting your changes ;)

strangely I didn't get conflicts at all. Probably because I moved the code you touched into a different file.

File: src/index.js [hunk 3]
```diff
@@ -1,452 +1,397 @@
...
+
+    this.onMessage = f => f;
+    this.promises.getUserId = null;
+    this.messageQueue = [];
+    this.onConnectionChangeCallback = f => f;
+    this._clearInitPromise();
+  };
 
-const setUpSocket = () => {
-  if (!store.notificationsChannelURL) {
```
Comment (abn) [inline src/index.js]: Is this maintained still? Or do we not need this anymore?

https://github.com/twyla-ai/widget-core/commit/1922d118172d29e2f54e6e92c8e922e5b42e7c36

File: src/utils.js (hunk 2 above)
Comment (arjunu) [reply src/utils.js]: changed, but I had to keep some of the other code to return project name and workspace name which are used for the name of the cookie.

File: src/tests/index.test.js (hunk 1 above)
Comment (arjunu) [reply src/tests/index.test.js]: removed

File: src/index.js (hunk 3 above)
Comment (arjunu) [reply src/index.js]: oops I missed this too. So this would be caught while parsing the hook URL right? So we don't need this.

**Pull request merged**